<launch>
  <arg name="input_cloud" default="/openni_camera/depth_registered/points" />
  <arg name="input_image" default="/openni_camera/rgb/image_rect_color" />
  <arg name="input_cluster" default="/multi_euclidean_clustering/output_indices" />
  <arg name="input_instance_boxes" default="/labeled_bounding_box_publisher/output/labeled_instance_boxes" />
  <arg name="input_cluster_boxes" default="/labeled_bounding_box_publisher/output/labeled_cluster_boxes" />
  <arg name="input_labels" default="/mask_rcnn_instance_segmentation/output/labels" />

  <arg name="labels" default="mask_rcnn_labels_191023.yaml" />
  <arg name="prefix" default="$(env HOME)/.ros/neatness_estimator" />

  <arg name="fe_service_topic" default="/objects_feature_extractor/extract" />
  <arg name="de_service_topic" default="/distance_estimator/estimate" />

  <node name="estimation_module_interface"
        pkg="nodelet" type="nodelet"
        args="standalone neatness_estimator/EstimationModuleInterface"
        output="screen">
    <remap from="~input_cloud" to="$(arg input_cloud)" />
    <remap from="~input_image" to="$(arg input_image)" />
    <remap from="~input_cluster" to="$(arg input_cluster)" />
    <remap from="~input_labels" to="$(arg input_labels)" />
    <remap from="~input_instance_boxes" to="$(arg input_instance_boxes)" />
    <remap from="~input_cluster_boxes" to="$(arg input_cluster_boxes)" />
    <rosparam command="load"
              file="$(find neatness_estimator)/models/$(arg labels)" />
    <rosparam subst_value="true">
      approximate_sync: true
      prefix: $(arg prefix)
      fe_service_topic: $(arg fe_service_topic)
      de_service_topic: $(arg de_service_topic)
      get_color_mask: true
    </rosparam>

  </node>

</launch>
